src/Ast/Type/TypeProjectionNode.php

```diff
@@ -0,0 +1,33 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\PhpDocParser\Ast\Type;
+
+use PHPStan\PhpDocParser\Ast\NodeAttributes;
+
+class TypeProjectionNode implements TypeNode
```

Comment:
Ideally `TypeProjectionNode` should not implement `TypeNode`, because it is not a *type* and cannot be used by places where *type* is expected. It's only allowed in place of *generic argument*.

Reply:
It is roughly equivalent to `CallableTypeParameterNode` which implements just `Node` instead of `TypeNode`.

Reply:
I made a suggestion here: https://github.com/phpstan/phpdoc-parser/pull/138#discussion_r901503448